src/main/java/uk/gov/hmcts/reform/sandl/snlevents/repository/queries/HearingQueries.java

```diff
@@ -0,0 +1,282 @@
+
+        return entityManager.createQuery(select);
+    }
+
+    private Predicate createWherePredicates(List<SearchCriteria> searchCriteriaList,
+                                            CriteriaBuilder cb,
+                                            CriteriaQuery<HearingSearchResponse> cq,
+                                            Root<Hearing> hearingRoot) {
+        Predicate restrictions = cb.conjunction();
+
+        for (SearchCriteria criteria: searchCriteriaList) {
+            ComparisonOperations operation = criteria.getOperation();
+            Predicate restriction = null;
+
+            if (criteria.getKey().equals("listingStatus")
+                && operation.equals(ComparisonOperations.EQUALS)) {
+
+                boolean isListed = criteria.getValue().toString().equals("listed");
+                restriction = createListingStatusPredicate(cb, cq, hearingRoot, isListed);
+            } else if (operation.equals(ComparisonOperations.EQUALS)) {
+                restriction = createEqualsPredicate(cb, hearingRoot, criteria);
+            } else if (operation.equals(ComparisonOperations.IN)) {
+                restriction = createInPredicate(hearingRoot, criteria);
+            } else if (operation.equals(ComparisonOperations.IN_OR_NULL)) {
+                restriction = createInOrNullPredicate(cb, hearingRoot, criteria);
+            } else if (operation.equals(ComparisonOperations.LIKE)) {
+                restriction = createLikePredicate(cb, hearingRoot, criteria);
+            } else {
+                throw new IllegalArgumentException("Hearing SearchCriteria keys or values not supported");
+            }
+            restrictions = cb.and(restrictions, restriction);
+        }
+
+        return restrictions;
+    }
+
+    private Predicate createLikePredicate(CriteriaBuilder cb, Root<Hearing> hearingRoot, SearchCriteria criteria) {
+        return cb.like(hearingRoot.get(criteria.getKey()), "%" + criteria.getValue().toString() + "%");
+    }
+
+    private Predicate createInOrNullPredicate(CriteriaBuilder cb, Root<Hearing> hearingRoot, SearchCriteria criteria) {
+        Predicate where;
+        String key = criteria.getKey();
+        Path<Object> rootKey = null;
+        if (key.contains(".")) {
+            rootKey = hearingRoot.get(key.split("\\.") [0]).get(key.split("\\.") [1]);
+        } else {
+            rootKey = hearingRoot.get(key);
+        }
+        where = cb.or(
+            rootKey.in(getArrayValues(key,
+                (List<String>) criteria.getValue())),
+            rootKey.isNull());
+        return where;
+    }
+
+    private Predicate createInPredicate(Root<Hearing> hearingRoot, SearchCriteria criteria) {
+        String key = criteria.getKey();
+        Path<Object> rootKey = null;
+        if (key.contains(".")) {
+            rootKey = hearingRoot.get(key.split("\\.") [0]).get(key.split("\\.") [1]);
+        } else {
+            rootKey = hearingRoot.get(key);
+        }
```

Comment:
Lines above and code in createInOrNullPredicate method could be extracted probably?